nevergrad/optimization/base.py

```diff
@@ -487,7 +489,7 @@ def recommend(self) -> p.Parameter:
             on the function (:code:`objective_function(*candidate.args, **candidate.kwargs)`).
         """
         recom_data = self._internal_provide_recommendation()  # pylint: disable=assignment-from-none
-        if recom_data is None:
+        if recom_data is None or any(np.isnan(recom_data)):
```

Comment:
if the optimization method returns nan then let us use our own recommendation.

Reply:
safer indeed